src/util/codec/mysql/time.rs

```diff
@@ -230,18 +235,28 @@ impl Time {
 
         let frac = try!(parse_frac(frac_str.as_bytes(), fsp));
         if y == 0 && m == 0 && d == 0 && h == 0 && minute == 0 && sec == 0 {
-            return Ok(zero_datetime());
+            return Ok(zero_datetime(tz));
         }
         if y < 0 || y > 9999 {
             return Err(box_err!("unsupport year: {}", y));
         }
-        let t = try!(ymd_hms_nanos(y, m, d, h, minute, sec, frac * TEN_POW[9 - fsp as usize]));
+        let t = try!(ymd_hms_nanos(tz,
+                                   y,
+                                   m,
+                                   d,
+                                   h,
+                                   minute,
+                                   sec,
+                                   frac * TEN_POW[9 - fsp as usize]));
         Time::new(t, types::DATETIME as u8, fsp as i8)
     }
 
-    pub fn from_packed_u64(t: u64, tp: u8, fsp: i8) -> Result<Time> {
+    /// Get time from packed u64. When `tp` is `TIMESTAMP`, the packed time should
+    /// be a UTC time; otherwise the packed time should be in the same timezone as `tz`
+    /// specified.
+    pub fn from_packed_u64(t: u64, tp: u8, fsp: i8, tz: &FixedOffset) -> Result<Time> {
```

Comment:
In this function, type of 't' has changed from u64 to NaiveDateTime, it's hard to read.
We should have different name pattern for packed u64 time, timezone aware datetime and naive datetime